docs/wiki/get-started/More-with-CSP.md

```diff
@@ -0,0 +1,141 @@
+
+```python
+class Trade(csp.Struct):
+    price: float
+    qty: int
+    buy: bool
+```
+
+## Build computing nodes
+
+To calculate volume-weighted averages, you need the cumulative sum of previous trade prices and quantities. Hence, your csp node needs to store stateful information. You can use \[`csp.state`\] to declare stateful variables that bound to the node.
+
+A csp node can return multiple named outputs, denoted as `csp.Outputs` type, created using the `csp.output` function. The individual return values can be accessed with dot notation.
+
+```python
+@csp.node
+def vwap(trade: ts[Trade]) -> csp.Outputs(vwap=ts[float], qty=ts[int]):
+    with csp.state():
+        s_cum_notional = 0.0
+        s_cum_qty = 0
+
+    if csp.ticked(trade):
+        s_cum_notional += trade.price * trade.qty
+        s_cum_qty += trade.qty
+
+        csp.output(vwap=s_cum_notional / s_cum_qty, qty=s_cum_qty)
+```
+
+```python
+@csp.node
+def calc_pnl(vwap_trade: ts[Trade], mark_price: ts[float]) -> ts[float]:
+    if csp.ticked(vwap_trade, mark_price) and csp.valid(vwap_trade, mark_price):
+        if vwap_trade.buy:
+            pnl = (mark_price - vwap_trade.price) * vwap_trade.qty
+        else:
+            pnl = (vwap_trade.price - mark_price) * vwap_trade.qty
+
+        return pnl
+```
+
+## Create workflow graph
+
+To create example `ask`, `bid`, and `trades` values, you can use [`csp.curve`](Base-Adapters-API#cspcurve).
+This commonly-used type in csp converts a list of (non-CSP) data into a ticking, csp-friendly inputs.
+
+```python
+@csp.graph
+def my_graph():
+    st = datetime(2020, 1, 1)
+
+    # Dummy bid/ask trade inputs
+    bid = csp.curve(
+        float,
+        [(st + timedelta(seconds=0.5), 99.0), (st + timedelta(seconds=1.5), 99.1), (st + timedelta(seconds=5), 99.2)],
+    )
+
+    ask = csp.curve(
+        float,
+        [
+            (st + timedelta(seconds=0.6), 99.1),
+            (st + timedelta(seconds=1.3), 99.2),
+            (st + timedelta(seconds=4.2), 99.25),
+        ],
+    )
+
+    trades = csp.curve(
+        Trade,
+        [
+            (st + timedelta(seconds=1), Trade(price=100.0, qty=50, buy=True)),
+            (st + timedelta(seconds=2), Trade(price=101.5, qty=500, buy=False)),
+            (st + timedelta(seconds=3), Trade(price=100.50, qty=100, buy=True)),
+            (st + timedelta(seconds=4), Trade(price=101.2, qty=500, buy=False)),
+            (st + timedelta(seconds=5), Trade(price=101.3, qty=500, buy=False)),
+            (st + timedelta(seconds=6), Trade(price=101.4, qty=500, buy=True)),
+        ],
+    )
+```
+
+The next step is to separate the buying and selling transactions that are captured in `Trade.buy` and calculate the VWAP. You can use the [csp.split](Base-Nodes-API#cspsplit) function for this. It splits input based on a boolean flag.
+
+> \[!TIP\]
+> To perform the opposite operation of a merge you can use [csp.merge](Base-Nodes-API#cspmerge).
```

Comment:
cant tell if this link is broken or the wiki magically figures out where this is.  If I look at the .md file in github's preview it shows that it links to get-started/Base-Nodes-API ( but its actually under api-references )

Reply:
The wiki itself is linear, so only the title matters. You can preview the links here: https://github.com/pavithraes/csp/wiki/More-with-CSP#create-workflow-graph